docs/design/resources.md

```diff
@@ -0,0 +1,120 @@
+# Operator Resources
+
+The etcd cluster operator extends the Kubernetes API with two new
+resource definitions:
+
+* `EtcdCluster`, or the "cluster resource", defines an entire
+  operating cluster.
+* `EtcdPeer`, or the "peer resource", defines a single peer of the
+  etcd cluster. This principally concerns a Pod, but also includes
+  attached storage and network identity.
+
+## The Peer Resource
+
+The peer resource represents a single peer in an etcd
+cluster. Encoding its bootstrap instructions, persistence settings,
+and extra pod settings (e.g., overridden CPU limits). Its `status`
+field exposes information about the peer including its hostname. In
+future the status may also include the liveness status if it's
+currently the leader of the etcd cluster.
+
+When reconciling the etcd peer resource the operator will do the
+following:
+
+1. If a correctly named persistent volume claim (PVC) does not exist,
```

Comment:
is it worth designing around the idea that we may eventually want to offer different storage mechanisms? e.g. people might find in-memory (or in-container) storage useful for experimentation. not suggesting we implement these right away but its worth considering when designing the API

Reply:
👍 as long as we keep this in mind when doing the API design